sample/host/src/App.tsx

```diff
@@ -0,0 +1,72 @@
+import { Route, useHoistedRoutes, useRoutes } from "@squide/react-router";
+import { RouterProvider, createBrowserRouter } from "react-router-dom";
+import { lazy, useCallback, useMemo } from "react";
+
+import { AuthenticationBoundary } from "./AuthenticationBoundary.tsx";
+import { RootErrorBoundary } from "./RootErrorBoundary.tsx";
+import { RootLayout } from "./RootLayout.tsx";
+
+const AuthenticatedLayout = lazy(() => import("./AuthenticatedLayout.tsx"));
+const ModuleErrorBoundary = lazy(() => import("./ModuleErrorBoundary.tsx"));
+
+const Home = lazy(() => import("./Home.tsx"));
+const Login = lazy(() => import("./Login.tsx"));
+
+export function App() {
+    const routes = useRoutes();
+
+    const wrapManagedRoutes = useCallback((managedRoutes: Route[]) => {
+        return {
+            // Pathless route to set a root layout and a root error boundary.
+            element: <RootLayout />,
+            errorElement: <RootErrorBoundary />,
+            children: [
+                {
+                    path: "/login",
+                    element: <Login />
+                },
+                {
+                    // Pathless route to set an authenticated boundary.
+                    element: <AuthenticationBoundary />,
+                    children: [
+                        {
+                            // Pathless route to set an authenticated layout.
+                            element: <AuthenticatedLayout />,
+                            children: [
+                                {
+                                    // Pathless route to set an error boundary inside the layout instead of outside.
+                                    // It's quite useful to prevent losing the layout when an unmanaged error occurs.
+                                    errorElement: <ModuleErrorBoundary />,
+                                    children: [
+                                        {
+                                            index: true,
+                                            element: <Home />
+                                        },
+                                        ...managedRoutes
+                                    ]
+                                }
+                            ]
+                        }
+                    ]
+                }
+            ]
+        };
+    }, []);
+
+    // Using the useHoistedRoutes hook allow routes hoisted by modules to be rendered at the root of the router instead of under the root layout.
```

Comment:
Is there a nice way we could wrap this instead of having having react-router-dom imports in the sample?